File: UniShare.API/Models/DTOs/Listings/ListingImageDto.cs
namespace UniShare.API.Models.DTOs.Listings;

public class ListingImageDto
{
    public Guid Id { get; set; }
    public string ImageUrl { get; set; } = null!;
    public int DisplayOrder { get; set; }
    public bool IsCover { get; set; }
}

File: UniShare.APP/lib/features/images/models/listing_image_dto.dart
import 'package:json_annotation/json_annotation.dart';

part 'listing_image_dto.g.dart';

@JsonSerializable()
class ListingImageDto {
  final String id;
  final String imageUrl;
  final bool isCover;
  final int displayOrder;

  const ListingImageDto({
    required this.id,
    required this.imageUrl,
    required this.isCover,
    required this.displayOrder,
  });

  factory ListingImageDto.fromJson(Map<String, dynamic> json) =>
      _$ListingImageDtoFromJson(json);

  Map<String, dynamic> toJson() => _$ListingImageDtoToJson(this);
}
